packages/api/src/command/medical/document/process-doc-query-webhook.ts

```diff
@@ -15,37 +15,86 @@ const { log } = out(`Doc Query Webhook`);
 const isSandbox = Config.isSandbox();
 const DOWNLOAD_WEBHOOK_TYPE = "medical.document-download";
 const CONVERSION_WEBHOOK_TYPE = "medical.document-conversion";
+type WebhookType = typeof DOWNLOAD_WEBHOOK_TYPE | typeof CONVERSION_WEBHOOK_TYPE;
+
+export const handleWebhookBeingSent = async ({
+  patient,
+  requestId,
+}: {
+  patient: Patient;
+  requestId: string;
+}) => {
+  const downloadStatus = patient.data?.documentQueryProgress?.download?.status;
+  const convertStatus = patient.data?.documentQueryProgress?.convert?.status;
+
+  const isDownloadFinished = downloadStatus === "completed" || downloadStatus === "failed";
+  const isConvertFinished = convertStatus === "completed" || convertStatus === "failed";
+
+  if (isDownloadFinished) {
+    await processDocQueryProgressWebhook({
+      patient,
+      requestId,
+      progressType: "download",
+      status: downloadStatus,
+    });
+  } else if (isConvertFinished) {
+    await processDocQueryProgressWebhook({
+      patient,
+      requestId,
+      progressType: "convert",
+      status: convertStatus,
+    });
+  }
+};
 
 /**
  * Processes the document query progress to determine if when to send the document download and conversion webhooks
  */
 export const processDocQueryProgressWebhook = async ({
   patient,
-  documentQueryProgress,
   requestId,
+  progressType,
+  status,
 }: {
   patient: Pick<Patient, "id" | "cxId" | "externalId">;
-  documentQueryProgress: DocumentQueryProgress;
   requestId: string;
+  progressType: ProgressType;
+  status: DocumentQueryStatus;
 }): Promise<void> => {
   const { id: patientId } = patient;
 
   try {
     const webhooks = await getAllWebhookRequestByRequestId(requestId);
     const webhookSet = new Set(webhooks.map(webhook => webhook.type));
-    const downloadWebhookSent = webhookSet.has(DOWNLOAD_WEBHOOK_TYPE);
-    const convertWebhookSent = webhookSet.has(CONVERSION_WEBHOOK_TYPE);
 
-    await handleDownloadWebhook(downloadWebhookSent, patient, requestId, documentQueryProgress);
-    await handleConversionWebhook(convertWebhookSent, patient, requestId, documentQueryProgress);
+    if (progressType === "convert") {
+      handleWebhook({
```

Comment:
I think we should avoid not awaiting for functions in multiple places... we're not awaiting here and not awaiting inside the function itself. Let's pick one, otherwise we can find ourselves not knowing what the state of the app is.

I'd defer to higher level calls as much as possible, like `getDocumentsFromCW` inside `queryDocumentsAcrossHIEs`... inside `getDocumentsFromCW` everything is awaited and we don't worry whether its being processed in parallel to CQ or not.